File: WoWSFT-Shared/src/main/kotlin/WoWSFT/model/gameparams/ship/component/torpedo/TorpedoAmmo.java
package WoWSFT.model.gameparams.ship.component.torpedo;

import WoWSFT.config.WoWSFT;
import WoWSFT.model.gameparams.TypeInfo;
import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.util.List;

@WoWSFT
@JsonIgnoreProperties(ignoreUnknown = true)
public class TorpedoAmmo
{
    private List<Object> additionalShips;
    private double alertDist;
    private double alphaDamage;
    private double alphaPiercingHE;
    private String ammoType;
    private double armingTime;
    private double bulletDiametr;
    private double burnProb;
    private double costCR;
    private double damage;
    private double damageUW;
    private double depth;
    private double directDamage;
    private double fallingTimeCoef;
    private double id;
    private List<String> ignoreClasses;
    private String index;
    @JsonAlias("isDeepWater")
    private boolean deepWater;
    private double maxDist;
    private String name;
    private String planeAmmoType;
    private double speed;
    private double splashArmorCoeff;
    private double splashCubeSize;
    private TypeInfo typeinfo;
    private boolean uwAbility;
    private double uwCritical;
    private double visibilityFactor;
    private double volume;
}

File: WoWSFT-Shared/src/main/kotlin/WoWSFT/model/gameparams/ship/component/torpedo/TorpedoAmmo.kt
package WoWSFT.model.gameparams.ship.component.torpedo

import WoWSFT.config.WoWSFT
import WoWSFT.model.gameparams.TypeInfo
import com.fasterxml.jackson.annotation.JsonAlias
import com.fasterxml.jackson.annotation.JsonIgnoreProperties

@WoWSFT
@JsonIgnoreProperties(ignoreUnknown = true)
class TorpedoAmmo
{
    var additionalShips = listOf<Any>()
    var alertDist = 0.0
    var alphaDamage = 0.0
    var alphaPiercingHE = 0.0
    var ammoType = ""
    var armingTime = 0.0
    var bulletDiametr = 0.0
    var burnProb = 0.0
    var costCR = 0.0
    var damage = 0.0
    var damageUW = 0.0
    var depth = 0.0
    var directDamage = 0.0
    var fallingTimeCoef = 0.0
    var id = 0L
    var ignoreClasses = listOf<String>()
    var index = ""
    @JsonAlias("isDeepWater")
    var deepWater = false
    var maxDist = 0.0
    var name = ""
    var planeAmmoType = ""
    var speed = 0.0
    var splashArmorCoeff = 0.0
    var splashCubeSize = 0.0
    var typeinfo = TypeInfo()
    var uwAbility = false
    var uwCritical = 0.0
    var visibilityFactor = 0.0
    var volume = 0.0
}